src/nwb_benchmarks/benchmarks/streaming_base.py

```diff
@@ -0,0 +1,101 @@
+"""Base template classes for basic benchmarks for NWB streaming read."""
+import fsspec
+import h5py
+import pynwb
+import remfile
+from fsspec.asyn import reset_lock
+
+
+class FileReadStreamingBase:
+    """Base class for basic benchmarks for streaming an NWB file from the DANDI archive."""
+    def setup(self):
+        # Random IBL raw data file; not that many groups
+        self.s3_url = "https://dandiarchive.s3.amazonaws.com/blobs/8c5/65f/8c565f28-e5fc-43fe-8fb7-318ad2081319"
+
+    def fsspec_no_cache(self):
+        reset_lock()
+        fsspec.get_filesystem_class("https").clear_instance_cache()
+        filesystem = fsspec.filesystem("https")
+
+        with filesystem.open(path=self.s3_url, mode="rb") as byte_stream:
+            with h5py.File(name=byte_stream) as file:
+                with pynwb.NWBHDF5IO(file=file, load_namespaces=True) as io:
+                    nwbfile = io.read()
+
+        # Must be done at this level since teardown occurs outside of repetitions
+        # reset_lock()
+        # fsspec.get_filesystem_class("https").clear_instance_cache()
```

Comment:
```suggestion
```

Whoops, I left this here by mistake; as habit now I just call it before instantiating any `fsspec` filesystem